src/rt/test/perf/dining_phil/verona_dining_phil.cc

```diff
@@ -0,0 +1,112 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#include "args.h"
+
+#include <cpp/when.h>
+#include <memory>
+#include <test/harness.h>
+
+struct Fork
+{
+  size_t uses{0};
+
+  void use()
+  {
+    ++uses;
+  }
+
+  ~Fork()
+  {
+    // Contains the uses == 0 case, due to copies during
+    // setup needing to be destroyed.
+    // Should look to make example copy free.
+    check(((HUNGER * 2) == uses) || (uses == 0));
+  }
+};
+
+cown_ptr<Fork> get_left(std::vector<cown_ptr<Fork>>& forks, size_t index)
+{
+  return forks[index];
+}
+
+cown_ptr<Fork> get_right(std::vector<cown_ptr<Fork>>& forks, size_t index)
+{
+  // This code handles tables by splitting the index into the table and then
+  // the philosopher index.  It wraps the fork around for the last philosoher
+  // on each table.
+  size_t table_size = NUM_PHILOSOPHERS / NUM_TABLES;
+  size_t table = index / table_size;
+  size_t next_fork = (index + 1) % table_size;
+  return forks[(table * table_size) + next_fork];
+}
+

```

Comment:
This is a lot more code than the pthread example.  For showing Verona in a good light, it would be nice if you could use the same code for accessing the left / right fork in a circular array of `T` and use `std::mutex` or `cown_ptr<Fork>` as `T` so that it's clearer what the differences are.